Cargo.toml

```diff
@@ -30,6 +30,7 @@ rand = "0.8.5"
 rayon = "1.8.0"
 serde = "1.0.188"
 snafu = "0.7.5"
+spin = { version = "0.9.8", features = ["mutex", "spin_mutex"] }
```

Comment:
Burn uses this because they also need to operate on embedded devices, but that's not a concern for us. I'd just use the standard Mutex available in Rust, instead of relying on an external crate.